tensorflow/core/ops/nn_ops.cc

```diff
@@ -2456,7 +2456,29 @@ REGISTER_OP("_ROCmFusedBatchNormActivationInference")
       using shape_inference::ShapeHandle;
       using shape_inference::DimensionHandle;
 
-      VLOG(-1) << "SetShapFn called for _ROCmFusedBatchNormActivation";
+      ShapeHandle x;
+      TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 4, &x));
```

Comment:
Is it possible to separate logic related to BN+A kernel fusion into a separate commit?

Reply:
will wait for the CI to come back (in case there are errors I need to fix) and then break down the changes into two commits and update the PR